real_estate/real_estate/doctype/property_unit/property_unit.py

```diff
@@ -20,3 +32,15 @@ def set_status(self, update=False, status=None, update_modified=False):
 
 		if update:
 			self.db_set('booking_status', self.booking_status, update_modified=update_modified)
+
+	def set_unit_template_booked_by(self, update=False, status=None, update_modified=False):
+		bookings = frappe.get_all('Property Booking Order', {'property_unit': self.name, 'docstatus': 1})
+
+		if bookings and self.booking_status == 'Booked':
+			booked_by = frappe.get_value('Property Booking Order', bookings[0].name, 'customer')
```

Comment:
no need to use `frappe.get_value`. you can include the customer in `booking = frappe.get_all(...)`. no need to call the db twice

Reply:
fixed